pkg/edvutils/edvutils.go

```diff
@@ -63,3 +74,98 @@ func Base58Encoded128BitToUUID(name string) (string, error) {
 
 	return storeNameUUID.String(), nil
 }
+
+// CheckIfURI checks if the given string is a valid URI.
+func CheckIfURI(str string) error {
+	_, err := url.ParseRequestURI(str)
+	if err != nil {
+		return fmt.Errorf(messages.InvalidURI, str)
+	}
+
+	return nil
+}
+
+// CheckIfArrayIsURI checks if every string in the given string array is a valid URI.
+func CheckIfArrayIsURI(arr []string) error {
+	for _, str := range arr {
+		if err := CheckIfURI(str); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// ValidateRawJWE returns an error if the given raw JWE is empty or has invalid alg fields.
+func ValidateRawJWE(rawJWE json.RawMessage) error {
+	if len(rawJWE) == 0 {
+		return errors.New(messages.BlankJWE)
+	}
+
+	jwe := models.JSONWebEncryption{}
+	if err := json.Unmarshal(rawJWE, &jwe); err != nil {
+		return err
+	}
+
+	if err := checkAlg(&jwe); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkAlg(jwe *models.JSONWebEncryption) error {
+	if jwe.B64ProtectedHeaders != "" && jwe.B64ProtectedHeaders != none {
+		foundAlg, err := checkAlgInProtectedHeader(jwe.B64ProtectedHeaders)
+		if err != nil {
+			return err
+		}
+
+		if foundAlg {
+			return nil
```

Comment:
It looks like you're only checking to see if `alg` exists - but don't you also need to make sure that it's not `none`?

Reply:
fixed

Reply:
It looks like you're still not checking to see if `alg` is `none` for the protected headers

Reply:
Oops, my bad - I missed it